src/main/java/org/apache/sling/jcr/repoinit/impl/AclUtil.java

```diff
@@ -203,6 +203,7 @@ public static void setPrincipalAcl(Session session, String principalName, Collec
         }
 
         final PrincipalAccessControlList acl = getPrincipalAccessControlList(acMgr, principal);
+        checkState(acl != null, "No PrincipalAccessControlList available for principal '" + principal + "'."); 
```

Comment:
maybe here we should even give out the path of the underlying user, but not sure how to get it from principal...

Reply:
@kwin , {{Principal}} just has a name.... but there exists an extension in _jackrabbit-api_ named {{org.apache.jackrabbit.api.security.principal.ItemBasedPrincipal}} that has a {{getPath()}} method. but principals exposed by {{PrincipalManager}} are not required to be of that type. if there exists a principal source (represented by a PrincipalProvider that contains principals not tied to a dedicated JCR item, they might just be simple {{Principal}}s. see _oak-exercise_ module and {{ExternalGroupPrincipal}} in _oak-auth-external_ for examples.

Reply:
Thanks for the hint, implemented in https://github.com/apache/sling-org-apache-sling-jcr-repoinit/pull/14/commits/1f5c6db769fd942515f90a9301e65f943f2bfd8a